PWA Kit 2.7.1-alpha.0 release

vcua-mobify opened the pull request:
This PR contains the prep work going into the PWA Kit 2.7.1 release

The rough changelist for v2.7.1 are:

* [Move the MRT reference app to the SDKs, so that we can verify Node support (#966)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/966)
* [BUG: Changed type of the phone number field to bring up numberic keyboard on mobile devices - W-9871940 (#1016)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1016)
* [Update Retail React App Page Designer integration README (#1041)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1041)
* [Implement updateCustomerPassword as no-op. (#1031)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1031)
* [Allow query hook parameters to be null (#1046](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1046)
* [Fix addresses not having preferred address first. (#1051)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1051)
* [Make mergeBasket Conditional More Robust (#1048)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1048)
* [commerce-sdk-react Decode pre-fetched token and save auth data in storage (#1052)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1052)
* [Prevent modal to open when it fails to add an item to cart (#1053)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1053)
* [Fix getConfig referencing config from incorrect location (#1049)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1049)
* [fix(template-retail-react-app): product-list refinements (#957)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/957)
* [update docs for shopper-experience scope (#1059)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1059)
* [Update lockfiles to reflect current version. (#1071)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1071)
* [commerce-sdk-react Implement remaining Shopper Baskets cache logic (#1070)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1070)
* [commerce-sdk-react Fix Shopper Baskets Test case (#1082)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1082)
* [feat(pwa-kit-dev): minor performance improvements and added comments (#974)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/974)
* [Update dependencies. (#1079)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1079)
* [Implement Cache Logic for Shopper APIs (Contexts/Customers/Login/Orders) (#1073)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1073)
* [remove site alias and locale from location.state.directedFrom path (#1065)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1065)
* [Fix Page Designer ImageWithText Link component (#1092)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1092)
* [split ssr build on local (#1155)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1155)
* [add suffix to ssr build files (#1157)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1157)
* [Added session bridge call to login for phased launch (#1159)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1159)
* [Fix multi-value query params being lost (#1150)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1150)
* [Replace invalid value for wrap property (#1179)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1179)
* [Snyk Security upgrades (#1169)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1169)

bendvc committed:
Update `develop` with `release-v2.7.0` (#1033)

* Starting release process for 2.7.0

* Update Create MRT credentials file (#1025)

* Release 2.7: fix order of hooks (#1027)

* Return early only after all of the hooks are called

* Bumper version

---------

Co-authored-by: Ben Chypak <bchypak@salesforce.com>

* Quick fix for einstein data (#1028)

* Quick fix for einstein data

* Bump Version

* Re-lock package-lock files with npm 8

* Update package-lock.json

* Update einstein.js

* Regenerate lock files (#1030)

* Regenerate lock files

* Bump version to 2.7.0-alpha.3

* Bump to 2.7.0 (#1032)

* Begin development on 2.8.0

---------

Co-authored-by: Adam Raya <adamraya@users.noreply.github.com>
Co-authored-by: Vincent Marta <vmarta@salesforce.com>

Commit:
Move the MRT reference app to the SDKs, so that we can verify eg. Node support (#966)

echessman committed:
BUG: Changed type of the phone number field to bring up numberic keyboard on mobile devices - W-9871940 (#1016)

Co-authored-by: Ben Chypak <bchypak@mobify.com>

johnboxall committed:
Update Retail React App Page Designer integration README (#1041)

Co-authored-by: Richard Sexton <rsexton404@users.noreply.github.com>

wjhsf committed:
Implement `updateCustomerPassword` as no-op. (#1031)

wjhsf committed:
Allow query hook parameters to be `null`. (#1046)

* Remove unused util.

* Allow query hook parameters to be `null`.

wjhsf committed:
Fix addresses not having preferred address first. (#1051)

* Fix addresses not having preferred address first.

* Include all addresses, not just preferred address twice.

* Correctly include preferred address.

bendvc committed:
Make `mergeBasket` Conditional More Robust (#1048)

* Update merge logic

* Update CHANGELOG.md

* Lint

* PR feedback

* Rename isNewlyRegisters to isNew for simplicity

* Lint

kevinxh committed:
[commerce-sdk-react] Decode pre-fetched token and save auth data in storage (#1052)

* add test

* decode jwt data

* lint

* Update packages/commerce-sdk-react/src/auth/index.ts

Co-authored-by: Will Harney <62956339+wjhsf@users.noreply.github.com>

* rename parseSlasJWT

* Update packages/commerce-sdk-react/src/auth/index.test.ts

Co-authored-by: Will Harney <62956339+wjhsf@users.noreply.github.com>

* convert all .thens to await

* make fake token in tests more clear

* lint

---------

Co-authored-by: Will Harney <62956339+wjhsf@users.noreply.github.com>

vmarta committed:
Prevent modal to open when it fails to add an item to cart (#1053)

* prevent modal to open when it fails to add an item to cart

* Update comment

---------

Co-authored-by: Alex Vuong <alex.vuong@salesforce.com>

bendvc committed:
Fix `getConfig` referencing config from incorrect location (#1049)

* Initial Commit

* Update CHANGELOG.md

* Revert some testing code

* Get test coverage back up

* Test build directory folder before proceeding.

breadadams committed:
fix(template-retail-react-app): product-list refinements  (#957)

* fix(template-retail-react-app): product-list refinements

+ Ensure that the `selectedFilters` provided to the `Refinements` value components is always an array. And update said value components (checkbox, color, radio & size) to handle always receiving an array.
+ Apply some loose equality checks, catering for cases where a refinement URL param is parsed as a number but the refinement `.value` is a stringified number.
+ Fix bug where non-multiple filters could not be unchecked by selecting themselves (e.g. clicking an already checked checkbox).
+ Fix bug where a refinement URL param is parsed as a number, but was handled as a string (with `.split()`).
+ Fix bug where radio refinements would still appear as checked after clearing the value via `SelectedRefinements`.
+ Deprecate unused proptypes in `SizeRefinements`

* feat(retail-react-app): update `CheckboxRefinements`

+ Adjust the `onChange` logic so that multiple checkbox refinements can be selected

---------

Co-authored-by: Ben Chypak <bchypak@mobify.com>

clavery committed:
update docs for shopper-experience scope (#1059)

* add docs for experience scope

* Add `merge_group` event to trigger GA builds

---------

Co-authored-by: Will Harney <62956339+wjhsf@users.noreply.github.com>
Co-authored-by: Adam Raya Navarro <arayanavarro@salesforce.com>

wjhsf committed:
Update lockfiles to reflect current version. (#1071)

adamraya committed:
[commerce-sdk-react] Implement remaining Shopper Baskets cache logic (#1070)

* Refactor Shopper Basket cache following new pattern

* Fix types

* Update packages/commerce-sdk-react/src/hooks/ShopperBaskets/cache.ts

Co-authored-by: Will Harney <62956339+wjhsf@users.noreply.github.com>

* PR Feedback

* Clean up unused utils and types after refactor

* Add missing response basketId to queryKey

* Implement cache for the remaining mutations

* Tests WIP

* PR Feedback & Use query key path in `deleteBasket` cache

* Add tests for mutations returning void response

* PR Feedback

* Remove unused `and` hooks util

* Revert "Remove unused `and` hooks util"

This reverts commit c0a364a4a39b2b05ed461efb392af62bc2130e80.

---------

Co-authored-by: Will Harney <62956339+wjhsf@users.noreply.github.com>

adamraya committed:
[commerce-sdk-react] Fix Shopper Baskets Test case (#1082)

* Fix Shopper Basket empty response test cases

* lint

ecRobertEngel committed:
feat(pwa-kit-dev): minor performance improvements and added comments (#974)

* feat(pwa-kit-dev): minor performance improvements and added comments

* docs(pwa-kit-dev): clean up comments

* refactor(pwa-kit-dev): update condition from PR feedback

---------

Co-authored-by: Will Harney <62956339+wjhsf@users.noreply.github.com>

wjhsf committed:
Update dependencies. (#1079)

* Remove internal packages to bypass lerna nonsense.

Include output.json to make restoring deps easier.

* Disable build scripts to make changing deps easier.

* Update root deps.

* Fix typescript issue.

* Create restore script to restore internal deps/scripts.

* Update commerce-sdk-react deps.

* Update test-commerce-sdk-react deps.

* Update template-typescript-minimal deps

* Improve restore script.

* Fix trailing comma.

* Update template-retail-react-app deps

* npm prune everything

* Update pwa-kit-runtime deps

* Update pwa-kit-react-sdk deps

* Update pwa-kit-create-app deps

* Update pwa-kit-dev deps (except eslint)

* Update pwa-kit-dev eslint deps

* Update internal-lib-build deps (except eslint)

* Update pwa-kit eslint deps

* Restore internal deps.

* Restore build scripts.

* Remove temporary helper files.

* Bump ua-parser-js to avoid vulnerable version.

* Anchor cross-env common dep to version used by internal-lib-build.

* Re-enable eslint.

bendvc committed:
Implement Cache Logic for Shopper APIs (Contexts/Customers/Login/Orders) (#1073)

* Initial commit

* Update packages/commerce-sdk-react/src/hooks/useAuthHelper.ts

Co-authored-by: Will Harney <62956339+wjhsf@users.noreply.github.com>

* Update packages/commerce-sdk-react/src/hooks/ShopperOrders/cache.ts

Co-authored-by: Will Harney <62956339+wjhsf@users.noreply.github.com>

* Update packages/commerce-sdk-react/src/hooks/ShopperOrders/cache.ts

Co-authored-by: Will Harney <62956339+wjhsf@users.noreply.github.com>

* Added root to all query keys, use remove over clear

* Remove previous impemented "clear" from utils

* Initial tests for shoppercontexts

* Update ShopperLogin tests

* Fix order tests

* Update packages/commerce-sdk-react/src/hooks/ShopperContexts/cache.ts

Co-authored-by: Will Harney <62956339+wjhsf@users.noreply.github.com>

* Update cache.ts

* Lint!

* Update Json.tsx

* Lint!

* Testing race condition in tests

* Re-add tests in other order.

* Update CHANGELOG.md

* Add todo to complete context cache work

* Update packages/commerce-sdk-react/src/hooks/ShopperBaskets/mutation.test.ts

Co-authored-by: Will Harney <62956339+wjhsf@users.noreply.github.com>

* Update packages/commerce-sdk-react/src/components/ShopperExperience/Page/index.tsx

Co-authored-by: Will Harney <62956339+wjhsf@users.noreply.github.com>

* Update useAuthHelper.ts

* Update packages/commerce-sdk-react/src/hooks/ShopperCustomers/cache.ts

Co-authored-by: Adam Raya <adamraya@users.noreply.github.com>

---------

Co-authored-by: Will Harney <62956339+wjhsf@users.noreply.github.com>
Co-authored-by: Adam Raya <adamraya@users.noreply.github.com>

sandragolden committed:
remove site alias and locale from location.state.directedFrom path (#1065)

* remove site alias and locale from location.state.directedFrom path

* moving call to removeSiteLocaleFromPath into use-navigation hook

* fixing failing test, added tests for removeSiteLocaleFromPath

* per code review, skipping failed test instead of using mocking

adamraya committed:
Fix Page Designer ImageWithText Link component (#1092)

* Fix PD ImageWithText Link component by using Chakra UI Link

* Use isAbsoluteURL to use react-router Link or Chakra Link component

* PR Feedback

* Clean up

* Update packages/template-retail-react-app/app/page-designer/assets/image-with-text/index.jsx

Co-authored-by: Ben Chypak <bchypak@mobify.com>

* Remove temporal page-viewer page to facilitate review

---------

Co-authored-by: Ben Chypak <bchypak@mobify.com>

kevinxh committed:
split ssr build on local (#1155)

kevinxh committed:
add suffix to ssr build files (#1157)

shethj committed:
Added session bridge call to login for phased launch (#1159)

* Added session bridge call to login for phased launch

* Fix code smell for session-bridge in hybrid

bendvc committed:
Fix multi-value query params being lost (#1150)

* Fix multi-value query params being lost

* Update CHANGELOG.md

vcua-mobify committed:
Snyk dependency updates (#1169)

* Dependency updates

* Update runtime package lock

* Bump cosmiconfig version to latest

shethj committed:
[Hybrid] PWA Kit should have a mechanism for replacing the access token when a SFRA login state is changed (#1171)

* Implement mechanism to store refresh token copy and compare with sfra

* Update tests and mocks for util function to check SFRA login state

* Fix linting issues

* FIx param types for util functionn

* Rename old isTokenValid to isTokenExpired

* Remove expiry for refresh_token in localstorage

* Update packages/template-retail-react-app/app/commerce-api/utils.js

Co-authored-by: Kevin He <kevin.he@salesforce.com>

* fix test

* Fix linting on use-auth-modal.test.js

* Update hasSFRAStateChanged logic to compare keys and values

* Fix linting

---------

Co-authored-by: Kevin He <kevin.he@salesforce.com>

vcua-mobify committed:
Add a redirect to login page after user signs out from checkout page (#1172)

* Add a redirect to login page after user signs out from checkout page

* Update CHANGELOG.md

* Remove history since navigate handles similarly

vcua-mobify committed:
Bump version number to 2.7.1-alpha.0

vcua-mobify committed:
Update changelogs

mdenchev-aiopsgroup committed:
#1174 Replace invalid value for wrap property (#1179)

vcua-mobify committed:
Merge branch 'develop' into 2.7.1-release

vcua-mobify committed:
Update changelogs

vcua-mobify committed:
Version bump to 2.7.1-preview.0

vcua-mobify committed:
Revert "Version bump to 2.7.1-preview.0"

This reverts commit 985a7e0c03df804e5c5d03a9577a84152b0cce73.

kevinxh commented:
It would be nice to see a compiled list of changes! For QA and communication purposes

vcua-mobify commented:
The changelist for v2.7.1 are:

* [Move the MRT reference app to the SDKs, so that we can verify Node support (#966)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/966)
* [BUG: Changed type of the phone number field to bring up numberic keyboard on mobile devices - W-9871940 (#1016)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1016)
* [Update Retail React App Page Designer integration README (#1041)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1041)
* [Implement updateCustomerPassword as no-op. (#1031)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1031)
* [Allow query hook parameters to be null (#1046](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1046)
* [Fix addresses not having preferred address first. (#1051)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1051)
* [Make mergeBasket Conditional More Robust (#1048)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1048)
* [commerce-sdk-react Decode pre-fetched token and save auth data in storage (#1052)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1052)
* [Prevent modal to open when it fails to add an item to cart (#1053)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1053)
* [Fix getConfig referencing config from incorrect location (#1049)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1049)
* [fix(template-retail-react-app): product-list refinements (#957)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/957)
* [update docs for shopper-experience scope (#1059)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1059)
* [Update lockfiles to reflect current version. (#1071)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1071)
* [commerce-sdk-react Implement remaining Shopper Baskets cache logic (#1070)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1070)
* [commerce-sdk-react Fix Shopper Baskets Test case (#1082)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1082)
* [feat(pwa-kit-dev): minor performance improvements and added comments (#974)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/974)
* [Update dependencies. (#1079)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1079)
* [Implement Cache Logic for Shopper APIs (Contexts/Customers/Login/Orders) (#1073)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1073)
* [remove site alias and locale from location.state.directedFrom path (#1065)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1065)
* [Fix Page Designer ImageWithText Link component (#1092)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1092)
* [split ssr build on local (#1155)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1155)
* [add suffix to ssr build files (#1157)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1157)
* [Added session bridge call to login for phased launch (#1159)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1159)
* [Fix multi-value query params being lost (#1150)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1150)
* [Replace invalid value for wrap property (#1179)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1179)
* [Snyk Security upgrades (#1169)](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1169)

vcua-mobify committed:
Merge branch 'release-2.7.x' into 2.7.1-release

vcua-mobify committed:
Update CHANGELOG.md

adamraya commented on package-lock.json:
```diff
@@ -1,17 +1,17 @@
     "": {
       "name": "pwa-kit",
-      "version": "2.7.0",
+      "version": "2.8.0-dev",
```
Something is wrong with the lock files, this shouldn't be `2.8.0-dev`.

vcua-mobify replied:
Before:
![Screenshot 2023-05-09 at 5 39 59 PM](https://github.com/SalesforceCommerceCloud/pwa-kit/assets/47404250/64406919-0da8-4311-8285-c7fd3d8eb9ac)

After:
![Screenshot 2023-05-09 at 5 56 31 PM](https://github.com/SalesforceCommerceCloud/pwa-kit/assets/47404250/0fab1d64-4283-4a20-af1a-b1583978cba5)

Thanks for catching that. I see it also. It's 2.7.1-alpha.0 at the top but 2.8.0-dev there.

I regenerated the lock files (and fixed an issue in ShopperLogin/mutation.test.js where an error is thrown if the project ends up building using commerce-sdk-isomorphic@1.10.1 vs commerce-sdk-isomorphic@1.9.0)

vcua-mobify replied:
This has been fixed.

vcua-mobify committed:
Rebuild lock files and fix ShopperLogin mutation test

vcua-mobify committed:
Revert "Rebuild lock files and fix ShopperLogin mutation test"

This reverts commit d9cfe509c8309c1b4083bad47b7483b8b8125511.

vcua-mobify committed:
Add additional properties to ShopperLogin test types (#1185)

kevinxh committed:
[V2] Re-generate lock files and fix hook lib tests (#1186)

* re-generate lock files and fix test

* Update packages/commerce-sdk-react/src/hooks/ShopperBaskets/index.test.ts

* Rebuild lockfiles using node 14 npm 8

* Revert "Rebuild lockfiles using node 14 npm 8"

This reverts commit 3d5c0cbe658d87a8f03fe887985cece0999c732c.

* Use pwa-kit-dev for lint and format

* Revert "Use pwa-kit-dev for lint and format"

This reverts commit f46d83eede0faa3c00d8fe816e40175d6c42af83.

* Add typescript to internal-lib-build and rebuild lock files

---------

Co-authored-by: vcua-mobify <47404250+vcua-mobify@users.noreply.github.com>
Co-authored-by: vcua-mobify <vcua@salesforce.com>

vcua-mobify committed:
[V2] Internal lib build typescript dev dependency (#1194)

* Move typescript to dev and peer dependency

* Update package lock file

vcua-mobify committed:
Merge branch 'develop' into 2.7.1-release

vcua-mobify committed:
Lockfile updates
